lib/services/build.ts

```diff
@@ -176,7 +176,17 @@ export class BuildService implements Project.IBuildService {
 					}
 				} else if(!settings.buildForiOSSimulator) {
 					let deviceIdentifier = settings.device ? settings.device.getIdentifier() : undefined;
-					provisionData = this.$identityManager.autoselectProvision(appIdentifier, settings.provisionTypes, deviceIdentifier).wait();
+					try {
+						provisionData = this.$identityManager.autoselectProvision(appIdentifier, [constants.ProvisionType.AdHoc], deviceIdentifier).wait();
+					} catch (error) {
+						this.$logger.warn("Cannot generate QR code because an applicable AdHoc provisioning profile is not available.");
+						let additionalInfo = error.message.split(os.EOL)[1];
+						if (additionalInfo) {
+							this.$logger.warn(additionalInfo);
```

Comment:
because this may be valuable information to the user like for example
Cannot use provision X because no valid certificates found
Which would be a reminder that the user imported the provision only and forgot about certificates